tests/rptest/tests/read_replica_e2e_test.py

```diff
@@ -175,18 +175,32 @@ def _bucket_delta(self, bu1: BucketUsage,
         else:
             return None
 
-    @cluster(num_nodes=6)
+    @cluster(num_nodes=7)
     @matrix(partition_count=[10])
-    def test_produce_is_forbidden(self, partition_count: int) -> None:
-
-        self._setup_read_replica(partition_count=partition_count)
+    def test_writes_forbidden(self, partition_count: int) -> None:
+        """
+        Verify that the read replica cluster does not permit writes,
+        and does not perform other data-modifying actions such as
+        removing objects from S3 on topic deletion.
+        """
+
+        self._setup_read_replica(partition_count=partition_count,
+                                 num_messages=1000)
         second_rpk = RpkTool(self.second_cluster)
         with expect_exception(
                 RpkException, lambda e:
                 "unable to produce record: INVALID_TOPIC_EXCEPTION: The request attempted to perform an operation on an invalid topic."
                 in str(e)):
             second_rpk.produce(self.topic_name, "", "test payload")
 
+        objects_before = sum(1 for _ in self.redpanda.s3_client.list_objects(
```

Comment:
nit: Could you use `len` here?

Reply:
list_objects is an iterator, so no len.  The `1 for _ in ` thing is a more efficient version for saying `len(list(the iterator))`

Reply:
Fair enough. I'll resolve those comments.